library/src/scala/tasty/util/TastyPrinter.scala

```diff
@@ -0,0 +1,641 @@
+package scala.tasty.util
+
+import scala.tasty.Tasty
+
+class TastyPrinter[T <: Tasty with Singleton](val tasty: T) {
+  import tasty._
+
+  def stringOfTree(tree: Tree)(implicit ctx: Context): String = {
+    implicit val buff: StringBuilder = new StringBuilder
+    visitTree(tree)
+    buff.toString()
+  }
+
+  def stringOfTypeTree(tree: MaybeTypeTree)(implicit ctx: Context): String = {
+    implicit val buff: StringBuilder = new StringBuilder
+    visitTypeTree(tree)
+    buff.toString()
+  }
+
+  def stringOfType(tpe: MaybeType)(implicit ctx: Context): String = {
+    implicit val buff: StringBuilder = new StringBuilder
+    visitType(tpe)
+    buff.toString()
+  }
+
+  def stringOfModifier(mod: Modifier)(implicit ctx: Context): String = {
+    implicit val buff: StringBuilder = new StringBuilder
+    visitModifier(mod)
+    buff.toString()
+  }
+
+  def stringOfConstant(mod: Constant)(implicit ctx: Context): String = {
+    implicit val buff: StringBuilder = new StringBuilder
+    visitConstant(mod)
+    buff.toString()
+  }
+
+  private def visitTree(x: Tree)(implicit buff: StringBuilder, ctx: Context): Unit = x match {
```

Comment:
I'd hope this code could be written more nicely. Make up a custom data type? We could add overloaded `+=` decorators to StringBuilder such that you can write

    buf += "Select(" += qualifier += "," += name += "," += optionally(signature) += ")"

Reply:
Changed